src/Text/Pandoc/Extensions.hs

```diff
@@ -132,6 +132,7 @@ data Extension =
     | Ext_multiline_tables    -- ^ Pandoc-style multiline tables
     | Ext_native_divs             -- ^ Use Div blocks for contents of <div> tags
     | Ext_native_spans            -- ^ Use Span inlines for contents of <span>
+    | Ext_native_numbering    -- ^ Use pandoc native numbering for figures and tables
```

Comment:
The doc-comment looks a little misleading. It sounds like pandoc has some kind of "native numbering", which it doesn't (yet). I suggest rephrasing this as "Use output format's native numbering for figures and tables".

Reply:
Probably also a good idea to indicate that it's only currently supported for odt output

Reply:
@jgm, we don't do that for `Ext_styles` though, so I don't think it's necessary here. It could potentially be useful for docx too down the line, and it doesn't seem like a good idea to write documentation that someone might (and probably will) forget to update later which will then lead to confusion, chaos, dogs and cats living together, mass hysteria.

Reply:
Ok, updated. No mention of odt though. I guess the correct thing to do (to keep things consistent) would be to specify which options apply to which formats, but that's out of scope for this work at least.

Reply:
At least it needs to be documented clearly in the manual that this extension only affects ODT.  One could make a case for documenting in the Haddocks as well -- otherwise someone using pandoc as a Haskell library might make incorrect assumptions.  Not sure how to balance this against the possibility of not updating the comment in the future.  I don't feel too strongly about it, as long as it's clear in the manual.

Reply:
Thinking ahead:  If we had an enumerated type for formats (which is somewhere in the issue tracker), we could add a typeclass `ImplementedFor Format Extension`, and then it would be clear from the types themselves what extensions worked for what formats.

Reply:
@jgm, uhh... what would `class ImplementedFor Format Extension` accomplish exactly? I mean, both `Format` and `Extension` would be enum types, I guess, so the only instance could in fact be `instance ImplementedFor Format Extension`, which does not seem very useful... Unless we get into dark type magic and start lifting data constructors into type algebra, but in that case I think type families will be more appropriate than typeclasses.

As a side note, there is support for type-level strings in the typechecker, so one could conceivably get away without using a enum type for `Format`. That is going to be somewhat tricky since it's impossible to bring a runtime string to the type level, for obvious reasons, but I think it's possible.

Reply:
yes, you're right, that wasn't well thought out.